### ISSUE TITLE
[ch-algorithms/Grover's Algorithm] 3.1.2 and 2.1.2

### ISSUE THREAD

**Issue by suryanshsrivastava** [qiskit-community/qiskit-textbook#1374]
### Describe the bug

```
backend = least_busy(provider.backends(filters=lambda x: x.configuration().n_qubits >= 3 and 
                                   not x.configuration().simulator and x.status().operational==True))
```
gives

`TypeError: '>=' not supported between instances of 'str' and 'int'`

### To Reproduce

1. Load IBM Q account 
```
from qiskit import IBMQ
IBMQ.save_account('MY_API_TOKEN')
provider = IBMQ.load_account()
provider = IBMQ.get_provider("ibm-q")
```
2. Run the code example in section 2.1.2 Experiment with Real Devices and 3.1.2 Experiment with Real Device.

### Expected behavior

Loads IBM Q account. Gets the least busy backend device 
and prints for example `Running on current least busy device:  ibmq_mumbai`

### Version info

{'qiskit-terra': '0.19.1', 'qiskit-aer': '0.10.2', 'qiskit-ignis': '0.7.0', 'qiskit-ibmq-provider': '0.18.3', 'qiskit-aqua': None, 'qiskit': None, 'qiskit-nature': None, 'qiskit-finance': None, 'qiskit-optimization': None, 'qiskit-machine-learning': None}

**Comment by suryanshsrivastava** [comment]
Can be fixed by typecasting `x.configuration().n_qubits` inside the lambda function as:
```
device = least_busy(provider.backends(filters=lambda x: int(x.configuration().n_qubits) >= 3 and 
                                   not x.configuration().simulator and x.status().operational==True))
print("Running on current least busy device: ", device)
```

**Comment by frankharkins** [comment]
Closed via #1375
